www/lobby/id.ts

```diff
@@ -1,15 +1,38 @@
 import { Router, Request, Response } from 'express';
 import { User } from '../../lib';
 import { Lobby } from '../../lib/lobby';
+import { validateJwt, validateLobbyJwt } from '../../utils/jwt';
 
 export const lobby_id_router = Router();
 
 // Join a lobby
 /**
  * Body Params: id, lobbyToken, refreshToken
  */
-lobby_id_router.post('/:id', async (req: Request, res: Response) => {
-  res.status(200).json('placeholder post');
+lobby_id_router.post('/:id', validateJwt, validateLobbyJwt, async (req: Request, res: Response) => {
+  const userId: string = req.body.id;
+  const lobbyId: string = req.params.id;
+  const decodedLobbyId: string | undefined = req.body.lobbyId;
+
+  //check if user and lobby id's are valid
+  const verified = await verifyIds(userId, lobbyId);
+  if (!verified) return res.status(404).json('User or Lobby not found in database').end();
+  const lobby = verified[1];
+
+  const isParticipant = lobby.containsParticipant(userId);
+
+  if (!isParticipant && lobbyId !== decodedLobbyId) {
+    return res.status(406).json('Lobby token is invalid').end();
+  }
+
+  if(!isParticipant) {
+    const added = await lobby.addUser(userId);
+    /** as of now, addUser should always return true
+     * this is just in case we want to handle the possibility that writing to the database fails
+    */
+    if (!added) return res.status(406).json('Error writing to database').end(); // not sure what error code to use, if it failed to write to database
```

Comment:
in general, you want to **avoid** optimizing early! i learned this from ppl at amazon who were much more senior than me.. but basically optimizing early leads to writing complicated code that never gets updated anyways to meet the optimization.. 

however, that being said, ill accept this bc there is no downside to having it! but something to keep in mind as you develop!

Reply:
Ok thanks for letting me know. I'll remember that in the future.